framework/src/utils/MeshBaseDiagnosticsUtils.C

```diff
@@ -66,4 +66,95 @@ checkNonConformalMesh(const std::unique_ptr<MeshBase> & mesh,
   }
   pl->unset_close_to_point_tol();
 }
+
+bool
+checkFirstOrderEdgeOverlap(const std::unique_ptr<Elem> & edge1,
+                           const std::unique_ptr<Elem> & edge2,
+                           std::vector<double> & intersection_coords,
+                           const Real intersection_tol)
+{
+  // check that the two elements are of type EDGE2
+  mooseAssert(edge1->side_type(*(edge1->side_index_range()).end()) == 0,
```

Comment:
why are you not using the element of edge1 and edge2?

Reply:
So I might have totally missed something, but when I looked through the documentation for elements and edges I didn't see any member variables for what type of element it was?

Reply:
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1Elem.html#ad06f8ac22978144723179eed42d10bae

Reply:
That makes more sense. I think my mistake was that I saw it was pure virtual and didn't see where it was implemented. Anyways just fixed it!